include/tf-psa-crypto/check_config.h

```diff
@@ -99,11 +99,6 @@
 #error "MBEDTLS_CMAC_C defined, but not all prerequisites"
 #endif
 
-#if defined(MBEDTLS_NIST_KW_C) && \
```

Comment:
Since we are targetting 4.0 release, which is a major number change, I don't fully agree with this change. Major release means that we can ask the user to change its build configuration, so I prefer to do things in the proper way, i.e:
- keep the check here, just replacing legacy symbols with PSA ones and
- don't modify `crypto_adjust_config_dependencies.h`.

I guess this is another question for @mpg since it's a more general design decision.

Reply:
In the "past" (that is, 3.x) we had a hybrid world where `MBEDTLS_` symbols were like "if you want to enable this symbol you'll need to figure out all its dependencies and enable them yourself" and `PSA_WANT` symbols where like "just ask for what you want and we'll handle dependencies for you" (one big reason for that is it support accelerators/drivers a lot better).

In 1.0 most of the public crypto feature symbols will be `PSA_WANT`. (Ideally that would be all of them, but we're keeping some legacy symbols for crypto mechanisms that are not in PSA yet, like NIST-KW here, or LMS.) So I think it makes sense for those few remaining legacy symbols to obey the same logic as the `PSA_WANT` symbols, rather than retaining a hybrid model just for them.

Reply:
Oh OK, that's the part that I was missing. Thanks for the explanation!